codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessageBuilders.java

```diff
@@ -0,0 +1,272 @@
+            return this;
+        }
+
+        public ConnectBuilder willRetain(boolean willRetain) {
+            this.willRetain = willRetain;
+            return this;
+        }
+
+        public ConnectBuilder hasUser() {
+            this.hasUser = true;
+            return this;
+        }
+
+        public ConnectBuilder hasPassword() {
+            this.hasPassword = true;
+            return this;
+        }
+
+        public ConnectBuilder username(String username) {
+            this.username = username;
+            return this;
+        }
+
+        public ConnectBuilder password(String password) {
+            this.password = password;
+            return this;
+        }
+
+        public MqttConnectMessage build() {
+            MqttFixedHeader mqttFixedHeader =
+                    new MqttFixedHeader(MqttMessageType.CONNECT, false, MqttQoS.AT_MOST_ONCE, false, 0);
+            MqttConnectVariableHeader mqttConnectVariableHeader =
+                    new MqttConnectVariableHeader(
+                            version.protocolName(),
+                            version.protocolLevel(),
+                            hasUser,
+                            hasPassword,
+                            willRetain,
+                            willQos.value(),
+                            willFlag,
+                            cleanSession,
+                            keepAliveSecs);
+            MqttConnectPayload mqttConnectPayload =
+                    new MqttConnectPayload(clientId, willTopic, willMessage, username, password);
+            return new MqttConnectMessage(mqttFixedHeader, mqttConnectVariableHeader, mqttConnectPayload);
+        }
+    }
+
+    public static final class SubscribeBuilder {
+
+        private List<MqttTopicSubscription> subscriptions;
+        private int messageId;
+
+        SubscribeBuilder() {}
+
+        public SubscribeBuilder addSubscription(MqttQoS qos, String topic) {
+            if (subscriptions == null) {
+                subscriptions = new ArrayList<MqttTopicSubscription>();
+            }
+            subscriptions.add(new MqttTopicSubscription(topic, qos));
+            return this;
+        }
+
+        public SubscribeBuilder messageId(int messageId) {
+            this.messageId = messageId;
+            return this;
+        }
+
+        public MqttSubscribeMessage build() {
+            MqttFixedHeader mqttFixedHeader =
+                    new MqttFixedHeader(MqttMessageType.SUBSCRIBE, false, MqttQoS.AT_LEAST_ONCE, false, 0);
+            MqttMessageIdVariableHeader mqttVariableHeader = MqttMessageIdVariableHeader.from(messageId);
+            MqttSubscribePayload mqttSubscribePayload = new MqttSubscribePayload(subscriptions);
+            return new MqttSubscribeMessage(mqttFixedHeader, mqttVariableHeader, mqttSubscribePayload);
+        }
+    }
+
+    public static final class UnsubscribeBuilder {
+
+        private List<String> topicFilters = new ArrayList<String>();
```

Comment:
@andsel should be final or created lazy when a `topicFilter` is added. Also consider if the default initialCapacity makes sense.